publisher/publisher.go

```diff
@@ -147,6 +148,9 @@ func SetCorrelationID(correlationID string) Option {
 
 // New creates a new service bus publisher
 func New(topicName string, opts ...ManagementOption) (*Publisher, error) {
+	ctx := context.Background()
```

Comment:
really this should come from the caller, but that changes the api.
so keeping it for a separate PR.

Reply:
created #29